src/main/Application.ts

```diff
@@ -315,38 +334,51 @@ export class Application {
     props: ICloudAuthenticationWindowProps,
     resolveUser: boolean = false,
   ): Promise<raas.user.IAccountResponse> {
-    return new Promise((resolve, reject) => {
-      const window = this.windowManager.createWindow({
-        type: 'cloud-authentication',
-        props,
-      });
-      const listener = (status: ICloudStatus) => {
-        if (status.kind === 'authenticated') {
-          this.cloudManager.removeListener(listener);
-          resolve(status.account);
-          // Close the window once we're authenticated
-          window.close();
-        } else if (status.kind === 'error') {
+    const { window, existing } = this.windowManager.createWindow({
+      type: 'cloud-authentication',
+      props,
+    });
+
+    let authPromise: Promise<raas.user.IAccountResponse>;
+    if (existing) {
+      window.focus();
+      // authPromise is set in the else clause
+      authPromise = (window as any).authPromise;
+    } else {
+      // Hacky way to avoid recreating the auth promise every time the window is focused.
+      (window as any).authPromise = authPromise = new Promise<
```

Comment:
@nirinchev ... why did we do this again? It's super hacky as the comment states and very potent for a future bug.

Reply:
If the `authPromise` variable gets renamed or another part of the code does not respect this implicit interface it's ripe for a runtime error.

Reply:
Sure, that's why there's a comment on both places it's being used - I wouldn't be worried about people renaming the property in one case and not the other. The reason I marked it as a hack wasn't because I add a runtime property on the window object but because it felt semantically incorrect to load the window object with that information. But at the time I couldn't find a better place to persist that information as the architecture didn't seem to follow the MVVM/MVC patterns I am familiar with.